include/scale/detail/jam_compact_integer.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <array>
+#include <concepts>
+#include <cstddef>
+#include <cstdint>
+
+#include <scale/outcome/outcome_throw.hpp>
+#include <scale/scale_error.hpp>
+#include <scale/types.hpp>
+
+namespace scale::detail {
+
+  /**
+   * Encodes any integer type to jam-compact-integer representation
+   * @tparam T integer type
+   * @tparam S output stream type
+   * @param value integer value
+   * @return byte array representation of value as jam-compact-integer
+   */
+  template <typename T, typename S>
+    requires(std::unsigned_integral<T> or std::is_same_v<T, CompactInteger>)
+  void encodeJamCompactInteger(T integer, S &s) {
+    size_t value;
```

Comment:
```suggestion
    uint64_t value;
```

Reply:
Why? `size_t` represents biggest unsigned integer type considering of platform.